x-pack/plugin/esql/src/main/java/org/elasticsearch/xpack/esql/parser/ExpressionBuilder.java

```diff
@@ -926,15 +923,48 @@ String unresolvedAttributeNameInParam(ParserRuleContext ctx, Expression param) {
     }
 
     @Override
-    public Expression visitMatchBooleanExpression(EsqlBaseParser.MatchBooleanExpressionContext ctx) {
-        if (Build.current().isSnapshot() == false) {
-            throw new ParsingException(source(ctx), "MATCH operator currently requires a snapshot build");
+    public Expression visitMatchOperatorExpression(EsqlBaseParser.MatchOperatorExpressionContext ctx) {
+        EsqlBaseParser.MatchOptionsContext optionsCtx = ctx.matchOptions();
+        Fuzziness fuzziness = null;
+        Double boost = null;
+        if (optionsCtx != null) {
+            fuzziness = visitFuzzinessExpression(optionsCtx.fuzzinessExpression());
+            boost = visitBoostExpression(optionsCtx.boostExpression());
         }
+
         return new MatchQueryPredicate(
```

Comment:
could we potentially use the `match` function instead of the predicate?
some additional work would be needed - to be able to specify that the function is used as an operator (for error messages) and to be able to pass options like fuzziness and boosting. 
Passing options would not be a syntax change, it's purely a change we make to the `match` function implementation.
Maybe we need to subclass the `match` function - I am not sure.

I think this ties in well with the idea that we can have both an operator and a function with the same functionality, but behind the scenes the operator is reduced to the function.
The added bonus is that we will not need to make significant changes to the `Verifier` and we do not need to rely on options as a string approach we have for `MatchQueryPredicate`.
The other advantage is that when we decide to enhance the `match` query (e.g. with support for `semantic_text`, compute engine evaluator etc), the `:` operator should also inherit any improvements and we don't have to duplicate work.

Reply:
Creating a function behind the scenes was my initial approach @ioanatia . I changed my mind though:
- Operators and functions currently have different number of optional named parameters. There is no direct support in functions for this.
- As functions are serialized / deserialized, this implies TransportVersion changes for new params being added.
- Param type resolution errors are a bit weird. We would have to say "the first param of the [:] operator" for referring to the field. We can of course change the type error for operator / functions.
- We would have to add some method to understand if it's a function or an operator for error messages.

In the end, I found a lot of logic similar to `if (isOperator)`, plus the TransportVersion changes for additional params.

You can explore the previous approach in 2ee261bf959afec9e72e4da02e6983aeaa67d616, before I changed my mind.

I'm happy to keep exploring that option, but I would do that as a separate PR so we ensure functions and operators can work as a dual entity.